tests/api/retrieval.test.js

```diff
@@ -0,0 +1,324 @@
+  });
+  afterAll(() => {
+    delete process.env.TOKEN_SECRET;
+  });
+
+  it("Should return 400 status code b/c formID is undefined", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: 1,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+
+    client.getSession.mockReturnValue(mockSession);
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "10",
+        formID: "",
+      },
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(400);
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ error: "Bad Request" }));
+  });
+
+  it("Should return 400 status code with Invalid paylaod value found maxRecords ", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: undefined,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+
+    client.getSession.mockReturnValue(mockSession);
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=1",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "11",
+        formID: "22",
+      },
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(400);
+    expect(JSON.parse(res._getData())).toEqual(
+      expect.objectContaining({ error: "Invalid paylaod value found maxRecords" })
+    );
+  });
+
+  it("Should return a list of form responses with 200 status code", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: 1,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    client.getSession.mockReturnValue(mockSession);
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=2",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "8",
+        formID: "22",
+      },
+    });
+    const expectedRow = {
+      template_id: 1,
+      email: "forms@cds.ca",
+      temporary_token: token,
+      active: true,
+    };
+
+    executeQuery.mockImplementation((client, sql) => {
+      if (
+        sql.includes(
+          "SELECT * FROM form_users WHERE template_id = ($1) and email = ($2) and active = true"
```

Comment:
Better to specify the fields to return than to request *

Reply:
`active` field.